include/crocoddyl/multibody/costs/com-position.hxx

```diff
@@ -81,15 +83,19 @@ template <typename Scalar>
 void CostModelCoMPositionTpl<Scalar>::set_referenceImpl(const std::type_info& ti, const void* pv) {
   if (ti == typeid(Vector3s)) {
     cref_ = *static_cast<const Vector3s*>(pv);
+    ResidualModelCoMPosition* residual = static_cast<ResidualModelCoMPosition*>(residual_.get());
+    residual->set_reference(cref_);
   } else {
     throw_pretty("Invalid argument: incorrect type (it should be Vector3s)");
   }
 }
 
 template <typename Scalar>
-void CostModelCoMPositionTpl<Scalar>::get_referenceImpl(const std::type_info& ti, void* pv) const {
+void CostModelCoMPositionTpl<Scalar>::get_referenceImpl(const std::type_info& ti, void* pv) {
   if (ti == typeid(Vector3s)) {
     Eigen::Map<Vector3s> ref_map(static_cast<Vector3s*>(pv)->data());
+    ResidualModelCoMPosition* residual = static_cast<ResidualModelCoMPosition*>(residual_.get());
```

Comment:
This is a duplicate of the previous comment in centroidal-momentum-cost

Reply:
Same comment: this code will be deprecated soon.

I am marking these conversation as resolved, unfortunately, I don't think we should waste our time by improving a code that will be deprecated :)

Reply:
I am okay with keeping the current implementation, since it would need be deprecated soon. 

In that case, the only thing to be done is to bind the references, r, rx, ru to residual.ref, r, rx, ru, so that there is no break in the API.